kernel/src/lib.rs

```diff
@@ -582,6 +582,13 @@ pub trait ParquetHandler: AsAny {
         physical_schema: SchemaRef,
         predicate: Option<PredicateRef>,
     ) -> DeltaResult<FileDataReadResultIterator>;
+
+    fn read_parquet_files_by_id(
```

Comment:
there's more discussion to be had about this, but if we do decide to go with a `flag` way of communicating that we're using `id`s, I think we could just make it another arg on the existing `read_parquet_files` function

Reply:
Yeah I think that's another way we can represent this. Could be a boolean flag or an C-style enum.

Any preference between function, boolean flag, and enum flag?

Given that internally, representing it as a flag is the way I went with, that indicates that other engines may want to do the same.